gems/aws-sdk-s3/lib/aws-sdk-s3/plugins/md5s.rb

```diff
@@ -47,7 +47,9 @@ def md5(value)
           end
 
           def update_in_chunks(digest, io)
-            while chunk = io.read(CHUNK_SIZE, buffer ||= "")
```

Comment:
Minor: why not just change `""` to `String.new`!

Reply:
Because the buffer is come up from nowhere and unused anywhere.  
So "The Boy Scout Rule" applied.

Reply:
Makes sense to me! If it's for sure unused, then it's okay to be removed.